generator/generator_godog_test.go

```diff
@@ -0,0 +1,117 @@
+package generator
+
+import (
+	"fmt"
+	"github.com/cucumber/godog"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+func iSendGETRequestTo(endpoint string) error {
+
+	matcher, err := regexp.MatchString("/store/\\{id}", endpoint)
+	if err != nil {
+		panic(err)
+	}
+	url := "http://localhost:8000/" + endpoint
+	data := strings.NewReader("")
+
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodGet, url, data)
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	if endpoint == "/store" && resp.StatusCode != 404 {
+		return fmt.Errorf("expected 404 received %d", resp.StatusCode)
+	} else if matcher && resp.StatusCode != 501 {
+		return fmt.Errorf("expected 501 received %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func iSendPOSTRequestToWithPayload(endpoint, payload string) error {
```

Comment:
POST requests almost always have a payload, so I think a better name would be:
```suggestion
func sendPostRequest(endpoint, payload string) error {
```